app/models/service_template_ansible_playbook.rb

```diff
@@ -146,13 +148,20 @@ def update_catalog_item(options, auth_user = nil)
       next unless config_info[action]
       info = config_info[action]
 
-      new_dialog = create_new_dialog(info[:new_dialog_name], job_template(action), info[:hosts]) if info[:new_dialog_name]
-      config_info[action][:dialog_id] = new_dialog.id if new_dialog
+      if new_dialog_required(info)
+        new_dialog = create_new_dialog(info[:new_dialog_name], job_template(action), info[:hosts])
+        config_info[action][:dialog_id] = new_dialog.id if new_dialog
+      end
 
-      next unless info.key?(:playbook_id)
-      tower, params = self.class.send(:build_parameter_list, "miq_#{name}_#{action}", description, info)
-      params[:manager_ref] = job_template(action).manager_ref
-      ManageIQ::Providers::EmbeddedAnsible::AutomationManager::ConfigurationScript.update_in_provider_queue(tower.id, params, auth_user)
+      if new_job_template_required(info, action)
+        single_config = extract_single_config(info, action)
+        config_info.deep_merge!(self.class.send(:create_job_templates, name, description, single_config, auth_user))
```

Comment:
Though right now retirement is the only action that may need a new job template, we should make it ready for reconfigure also. `create_job_templates` is for multiple actions.

I would suggest to have the following logic
1. separate config_info to two hashes, one for updating job templates, and the other for new job templates
2. update job templates (and delete job templates if possible)
3. create new job templates
4. create new dialogs (and also delete old dialogs if possible)
5. call `super` to update other info.